### PR TITLE
Remove error when test dataloader used in test

### LINKED ISSUE #1496: Incorrect MisconfigurationException for models without dataloaders.
## 🐛 Bug

I have a model that does not have train, val and test dataloaders defined internally (it's a production system and it doesn't really make sense to have dataloaders). If I try to run `fit()` on it by passing in `train_dataloader` and `val_dataloaders`, it raises 
```python
pytorch_lightning.utilities.exceptions.MisconfigurationException: You have defined `test_step()`, but have not passed in a `test_dataloader()`.
```
This means that it's now impossible to train a model without dataloaders defined, as there's no way of passing in test dataloaders. I believe this was caused by this PR: https://github.com/PyTorchLightning/pytorch-lightning/pull/1434. This is happening at the tip of master.

### To Reproduce

1. Checkout the master branch
2. Define a model without data loaders
3. Run `fit()` with `train_dataloader` and `val_dataloaders`
4. See the exception

#### Code sample
```python
# MyModel doesn't have train_dataloader, val_dataloader or test_dataloader
model = MyModel()
trainer = pl.Trainer()
trainer.fit(model, train_dataloader=train, val_dataloaders=val)  # exception raised here
trainer.test(test_dataloaders=test)
```

### Expected behavior

There should be no exception raised during `fit()`.

### Environment
```
cuda:
	GPU:
	available:           False
	version:             None
packages:
	numpy:               1.18.1
	pyTorch_debug:       False
	pyTorch_version:     1.4.0
	pytorch-lightning:   0.7.4-dev
	tensorboard:         2.1.1
	tqdm:                4.43.0
system:
	OS:                  Darwin
	architecture:
		64bit
		
	processor:           i386
	python:              3.7.3
	version:             Darwin Kernel Version 18.7.0: Mon Feb 10 21:08:45 PST 2020; root:xnu-4903.278.28~1/RELEASE_X86_64
```

### PR DESCRIPTION
**Pull request by williamFalcon**
When we removed test dataloader we forgot to remove the error check

Fixes #1496

### CODE REVIEW

**Comment by williamFalcon** [comment]
@Borda Trail is not intuitive. Please leave as ```EvaluationTemplateModel```

**Comment by Borda** [comment]
> @Borda Trail is not intuitive. Please leave as `EvaluationTemplateModel`

ok, we just need to separate and we cannot use work test
- test as a phase from a model live cycle
- test as our case in testing
so can I have just `TemplateModel` because in this moment the evaluation is associated with the test as a model live cycle?

File: pytorch_lightning/trainer/trainer.py [hunk 1]
```diff
@@ -939,10 +939,11 @@ def test(self, model: Optional[LightningModule] = None, test_dataloaders: Option
         self.testing = True
 
         if test_dataloaders is not None:
-            if model is not None:
```
Comment (williamFalcon) [inline pytorch_lightning/trainer/trainer.py:942]: this way is much clearer. i  prefer clarity  over  fewer  lines

File: CHANGELOG.md [hunk 2]
```diff
@@ -111,6 +111,7 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
 ### Removed
 
+- Removed test for no test dataloader in .fit ([#1495](https://github.com/PyTorchLightning/pytorch-lightning/pull/1495))
```
Comment (Borda) [inline CHANGELOG.md:114]:
```suggestion
- Removed test for no test dataloader in `.fit()` ([#1495](https://github.com/PyTorchLightning/pytorch-lightning/pull/1495))
```

**Pull request merged**